### PR TITLE
(feat) O3-2391: Ability to hide observations in encounters via configuration

### PR DESCRIPTION
**Pull request by AlexanderMizgirev**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary
Sometimes it is necessary to hide something in the list of observations when viewing an encounter. For example, this could be lines about added attachments (see screenshot). It could also be other entities that do not need to be shown in the user interface.

## Screenshots
image (2)

## Related Issue
https://issues.openmrs.org/browse/O3-2391

### CODE REVIEW

**Comment by AlexanderMizgirev** [comment]
> (feat) O3-2391: Ability to hide observations in encounters via configuration

Fixes according to the documentation were applied
@ibacher

File: packages/esm-patient-chart-app/src/visit/visits-widget/encounter-observations/encounter-observations.component.tsx [hunk 1]
```diff
@@ -25,9 +27,12 @@ const EncounterObservations: React.FC<EncounterObservationsProps> = ({ observati
   }
 
   if (observations) {
+    const filteredObservations = observations?.filter((it) => {
+      return !hideFieldsByConceptUuid.includes(it?.concept?.uuid);
+    });
```
Comment (ibacher) [inline packages/esm-patient-chart-app/src/visit/visits-widget/encounter-observations/encounter-observations.component.tsx:32]:
```suggestion
    const filteredObservations = hideFieldsByConceptUuid ? observations?.filter((it) => {
      return !hideFieldsByConceptUuid.includes(it?.concept?.uuid) : observations;
    });
```

**Review by ibacher — COMMENTED** [review]
Looks good with one small add-on so that we don't do any work in the case where no concept UUIDs are set.

File: packages/esm-patient-chart-app/src/visit/visits-widget/encounter-observations/encounter-observations.component.tsx (hunk 1 above)
Comment (denniskigen) [reply packages/esm-patient-chart-app/src/visit/visits-widget/encounter-observations/encounter-observations.component.tsx:32]: Think you meant to go with: 

```tsx
const filteredObservations = hideFieldsByConceptUuid
  ? observations?.filter((it) => {
      return !hideFieldsByConceptUuid.includes(it?.concept?.uuid);
    })
  : observations;
```

File: packages/esm-patient-chart-app/src/config-schema.ts [hunk 2]
```diff
@@ -103,6 +103,14 @@ export const esmPatientChartSchema = {
       _description: 'The organization name displayed when image is absent',
     },
   },
+  hideFieldsByConceptUuid: {
+    _type: Type.Array,
+    _elements: {
+      _type: Type.ConceptUuid,
+    },
+    _description: 'An array of concept UUIDs that will be hidden in the observations output when viewing an encounter',
```
Comment (denniskigen) [inline packages/esm-patient-chart-app/src/config-schema.ts:111]:
```suggestion
    _description: 'An array of concept UUIDs. When this property is set to `true`, any observations in the observations list view that match any of the UUIDs in this array will be hidden.'
```

**Comment by elimm** [comment]
@ibacher @denniskigen I've recreated this [PR](https://github.com/openmrs/openmrs-esm-patient-chart/pull/1353) with all suggestions, the reason is @AlexanderMizgirev is not available any more to apply any changes, so please review the new one

**Comment by denniskigen** [comment]
Closing in favour of https://github.com/openmrs/openmrs-esm-patient-chart/pull/1353.

**Pull request closed without merging**